k4GaudiPandora/src/DDPandoraPFANewAlgorithm.cc

```diff
@@ -0,0 +1,489 @@
+  const std::vector<const edm4hep::CalorimeterHitCollection*>& hCalCollections,
+  const std::vector<const edm4hep::CalorimeterHitCollection*>& mCalCollections,
+  const std::vector<const edm4hep::CalorimeterHitCollection*>& lCalCollections,
+  const std::vector<const edm4hep::CalorimeterHitCollection*>& lhCalCollections,
+  const std::vector<const edm4hep::CaloHitSimCaloHitLinkCollection*>& caloLinkCollections
+) const {
+  try {
+    debug() << "DDPandoraPFANewAlgorithm - Run " << endmsg;
+    //(void)m_pandoraToLCEventMap.insert(PandoraToLCEventMap::value_type(m_pPandora, pLCEvent));
+
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, m_pDDMCParticleCreator->CreateMCParticles(MCParticleCollections));
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, m_pTrackCreator->CreateTrackAssociations(kinkCollections, prongCollections, splitCollections, v0Collections));
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, m_pTrackCreator->CreateTracks(trackCollections));
+    PANDORA_THROW_RESULT_IF(
+        pandora::STATUS_CODE_SUCCESS, !=,
+        m_pDDMCParticleCreator->CreateTrackToMCParticleRelationships(trackerHitLinkCollections, m_pTrackCreator->GetTrackVector(), trackCollections));
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, 
+      m_pCaloHitCreator->CreateCaloHits(eCalCollections, hCalCollections, mCalCollections, lCalCollections, lhCalCollections));
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+                            m_pDDMCParticleCreator->CreateCaloHitToMCParticleRelationships(
+                              caloLinkCollections, m_pCaloHitCreator->GetCalorimeterHitVector(), 
+                              eCalCollections, hCalCollections, mCalCollections, lCalCollections, lhCalCollections));
+
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, PandoraApi::ProcessEvent(*m_pPandora));
+
+    edm4hep::ClusterCollection               pClusterCollection;
+    edm4hep::ReconstructedParticleCollection pReconstructedParticleCollection;
+    edm4hep::VertexCollection                pStartVertexCollection;
+
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, 
+      m_pDDPfoCreator->CreateParticleFlowObjects(
+        pClusterCollection, pReconstructedParticleCollection, pStartVertexCollection,
+        MCParticleCollections, trackCollections, eCalCollections, hCalCollections, mCalCollections, lCalCollections, lhCalCollections));
+
+    PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, PandoraApi::Reset(*m_pPandora));
+    this->Reset();
+
+    return std::make_tuple(
+      std::move(pClusterCollection), 
+      std::move(pReconstructedParticleCollection), 
+      std::move(pStartVertexCollection)
+    );
+  } catch (pandora::StatusCodeException& statusCodeException) {
+    error() << "Marlin pandora failed to process event: " << statusCodeException.ToString() << endmsg;
```

Comment:
```suggestion
    error() << "Gaudi pandora failed to process event: " << statusCodeException.ToString() << endmsg;
```
Or at least something that doesn't throw people for a loop here ;)